test/connections.js

```diff
@@ -3,6 +3,34 @@ const dgram = require('dgram')
 const { swarm } = require('./helpers')
 const DHT = require('../')
 
+test('createServer + connect - destroy nodes without ending socket', { timeout: 60000 }, async function (t) {
+  const [a, b] = await swarm(t)
+  const lc = t.test('socket lifecycle')
+
+  t.plan(2)
+  lc.plan(2)
+
+  const server = a.createServer(function (socket) {
+    lc.pass('server side opened')
```

Comment:
you have to error handle the socket here, so it doesn't crash (ie attach an error listener that does nothing)